### PR TITLE
Add Exec and Query methods to FakePDO

### PR DESCRIPTION
**Pull request by M1ke**
Adds other public methods (rather than just `prepare()` to `FakePdo`)

The current `master` branch state of Psalm surfaces the following fails:

```
ERROR: ArgumentTypeCoercion - src/Parser/SQLLexer.php:24:13 - Argument 4 of preg_split expects int(0)|int(1)|int(2)|int(3)|int(4)|int(5)|int(6)|int(7), parent type positive-int provided (see https://psalm.dev/193)
            \PREG_SPLIT_DELIM_CAPTURE | \PREG_SPLIT_NO_EMPTY | \PREG_SPLIT_OFFSET_CAPTURE


ERROR: InvalidArgument - src/Parser/SQLLexer.php:32:44 - Argument 1 of Vimeo\MysqlEngine\Parser\SQLLexer::groupComments expects list<array{string, int}>, list<array{string, int}|string> provided (see https://psalm.dev/004)
            $tokens = $this->groupComments($tokens);


ERROR: InvalidArgument - src/Parser/SQLLexer.php:35:51 - Argument 1 of Vimeo\MysqlEngine\Parser\SQLLexer::groupEscapeSequences expects list<array{string, int}>, list<array{string, int}|string> provided (see https://psalm.dev/004)
            $tokens = $this->groupEscapeSequences($tokens);


ERROR: InvalidArgument - src/Parser/SQLLexer.php:37:41 - Argument 1 of Vimeo\MysqlEngine\Parser\SQLLexer::groupQuotedTokens expects list<array{string, int}>, list<array{string, int}|string> provided (see https://psalm.dev/004)
        return $this->groupQuotedTokens($tokens);
```

This branch does not resolve these, but also does not add new fails.

### CODE REVIEW

File: src/FakePdo.php [hunk 1]
```diff
@@ -97,20 +100,62 @@ public function lastInsertId($seqname = null) : string
...
+	 */
+	public function exec($statement)
+	{
+		$statement = trim($statement);
+		if (strpos($statement, 'SET ')===0){
+			return false;
+		}
+
+		$sth = $this->prepare($statement);
+		if ($sth->execute()){
```
Comment (M1ke) [inline src/FakePdo.php:142]: I believe this is the correct way to replicate the internal logic

File: src/FakePdo.php [hunk 2]
```diff
@@ -97,20 +100,62 @@ public function lastInsertId($seqname = null) : string
...
+	 * @param int $mode
+	 * @param null $arg3
+	 * @param array $ctorargs
+	 * @return Php7\FakePdoStatement|Php8\FakePdoStatement
+	 */
+	public function query($statement, $mode = PDO::ATTR_DEFAULT_FETCH_MODE, $arg3 = null, array $ctorargs = [])
+	{
+		$sth = $this->prepare($statement);
+		$sth->execute();
+		return $sth;
```
Comment (M1ke) [inline src/FakePdo.php:159]: Not certain if anything new is needed here to replicate internal logic but this does run properly

**Comment by M1ke** [comment]
So the fail on the build is coming from the `query` declaration where I've declared:

```
query($statement, $mode = PDO::ATTR_DEFAULT_FETCH_MODE, $arg3 = null, array $ctorargs = [])
```

Whereas the tests run on PHP 8 which requires this to be typed as 

```
query(string $statement, ?int $mode = null, mixed ...$fetchModeArgs)
```

But if we do that it will break on anything <PHP8. I wonder if this requires the tests to be run on the min supported version (PHP 7.1 according to composer.json)?

**Comment by muglug** [comment]
Thanks!

Please merge/rebase master! I've introduced separate `FakePdo` classes for PHP 7 & 8, allowing you to use the different APIs without breaking everything.

**Comment by M1ke** [comment]
So the last Psalm build on actions failed due to a Docker error... to make it build again I pushed some new `.gitignore` lines - they are relevant but obviously not directly to this PR so I can remove them if needed.

But on the plus side, all checks now pass with the split out PHP7/8 classes!

I'll update the Codeception branch at https://github.com/Codeception/module-db/pull/20 to use a specific class too.

**Comment by muglug** [comment]
Thanks!

**Pull request merged**